packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -159,6 +169,7 @@ export function toBindingSyncConfig(config: SyncConfiguration): binding.SyncConf
     customHttpHeaders,
     flxSyncRequested: !!flexible,
     ...parseClientResetConfig(clientReset, onError),
+    cancelWaitsOnNonfatalError: cancelWaitsOnNonFatalError,
```

Comment:
I agree with @elle-j here - I was actually referring to `Nonfatal` vs `NonFatal` too, we should pick one or upstream an issue / PR to rename it in core if we don't like the name.

Reply:
`Nonfatal` is not grammatically wrong, but I agree that `NonFatal` reads better, we also tend to use `non-<some word>` in our docs so using uppercase `F` seems consistent with that at least. Regarding Core, they use different spellings around the code base (e.g. `non_null`, `nonnull`, `NonNull`).

Reply:
I don't care either way, I think they should just be the same externally as internally to avoid confusion (which is more important than a grammar nit in my opinion).

Reply:
Mmmh, I don't know if I agree with this. I think that the core team has much more leeway in the naming of the variables, given that they don't need to expose those to developers, and can more or less decide internally how they want to do it. 
Instead we should try to keep somehow consistent how we want to name our variables and should not depend on how core wants to do it. The reason why we have this "problem" is because the binding generator is creating the variable names based on core, but that's something that we decided to put in between. Besides, we are not even forced to keep the same name that is coming from core (like Swift, that decide to change it). In this case I thought the name made sense (even though it's kinda long) and I just kept it more or less.

Reply:
Your decision 🙂